app/lib/features/chat_ng/providers/notifiers/reply_messages_notifier.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:acter/common/providers/chat_providers.dart';
+import 'package:acter/features/chat_ng/models/reply_message_state.dart';
+import 'package:acter/features/chat_ng/providers/chat_room_messages_provider.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::chat::reply_notifier');
+
+class ReplyMessageNotifier
+    extends AutoDisposeFamilyAsyncNotifier<ReplyMsgState, ReplyMsgInfo> {
+  Future<void> retryLoad() async {
+    state = const AsyncValue.loading();
+    state = await AsyncValue.guard(() => build(arg));
+  }
+
+  @override
+  Future<ReplyMsgState> build(ReplyMsgInfo arg) async {
+    // Watch messages to react to changes
+    final messages = ref.watch(
+      chatStateProvider(arg.roomId).select((value) => value.messageList),
+    );
+
+    // First try to find message in current state
+    final messageId = messages.firstWhere(
+      (eventId) => eventId == arg.originalId,
+      orElse: () => '',
+    );
+
+    if (messageId.isNotEmpty) {
```

Comment:
Please use types to indicate state, not the content if it ;)
```suggestion
    final String? messageId = messages.firstWhere(
      (eventId) => eventId == arg.originalId,
      orElse: () => null,
    );

    if (messageId != null) {
```

Reply:
I can conclude based on your below comment that this isn't needed at all. My thinking was to have that message fetched already if it's available in current state but since we providers are cached, this doesn't seem to have any noticeable difference in rendering speed, even if removed.